### PR TITLE
Enhancing tests for design module's total cost

### PR DESCRIPTION
**Pull request by nRiccobo**
Updated several design tests to include some kind of "test_total_cost". A simple test like this should capture a snapshot of the design module's total cost for given a base case. Further, this will aid in tracking if any logic, default costs, or cost algorithms change between additions. 

Additionally, I addressed some potential if-statements matching issues in electrical_design and mooring_system_design that @rafmudaf noticed in #171.

### CODE REVIEW

**Comment by rafmudaf** [comment]
Thanks @nRiccobo, nicely done.

Just a comment - using the form `"key" in self.config["key"].lower()` instead of `self.config["key"].lower() == "key"` means that ORBIT will support inputs that have the right parameter somewhere within the string such as `"mooring_system_design": {"mooring_type": ["SemiTautttt"]}`. That kind of flexibility can be really nice, but it's not the usual so I just wanted to point it out in case it wasn't intentional. You'll also have to be careful not to add a valid config value that appends something to an existing value like "HVDC_aluminum" etc for the cable.

Also, you could consider making all the inputs lower case in `load_config` and then use lower case in all the if-statements, and this would remove `.lower()` everywhere else. It's probably a painful amount of work to catch this everywhere, though.

Just my two cents. Thanks for addressing my other comment @nRiccobo!

File: ORBIT/phases/design/mooring_system_design.py [hunk 1]
```diff
@@ -161,15 +161,15 @@ def calculate_line_length_mass(self):
         """
 
         # Add extra fixed line length for drag embedments
-        if self.anchor_type == "Drag Embedment":
+        if any(w in self.anchor_type.lower() for w in ["drag", "embedment"]):
```
Comment (RHammond2) [inline ORBIT/phases/design/mooring_system_design.py:164]: It seems like some input standardization might be needed to reduce some of the work in this PR

**Review by RHammond2 — COMMENTED** [review]
Thanks for getting these additional checks in place!

I think you opened up a small can of worms with some of input handling methods, otherwise the only thing I'd add is a total design cost for the default fixed and floating scenarios to ensure all the costs are double checked.

File: ORBIT/phases/design/electrical_export.py [hunk 2]
```diff
@@ -274,7 +274,7 @@ def compute_number_cables(self):
         num_required = np.ceil(self._plant_capacity / self.cable.cable_power)
         num_redundant = self._design.get("num_redundant", 0)
 
-        if "HVDC" in self.cable.cable_type:
+        if "HVDC" in self.cable.cable_type.upper():
```
[the remark this answers is not in the document]
Comment (nRiccobo) [reply ORBIT/phases/design/electrical_export.py]: Added in latest commit.

File: ORBIT/phases/design/electrical_export.py [hunk 3]
```diff
@@ -539,7 +541,7 @@ def calc_substructure_mass_and_cost(self):
 
         substructure_pile_mass = (
             0
-            if "Floating" in self.substructure_type
+            if self.substructure_type.lower() == "floating"
```
[the remark this answers is not in the document]
Comment (nRiccobo) [reply ORBIT/phases/design/electrical_export.py]: Went back to the original form, but changed self.substructure_type to use a "title()" case format when first instantiated.

File: ORBIT/phases/design/mooring_system_design.py (hunk 1 above)
Comment (nRiccobo) [reply ORBIT/phases/design/mooring_system_design.py]: Went back to old logic, but changed self.anchor_type to "title()" case when first instantiated.

File: ORBIT/phases/design/mooring_system_design.py [hunk 4]
```diff
@@ -209,7 +213,7 @@ def calculate_line_length_mass(self):
                 + self.rope_length * rope_mass_per_m
             ) / 1e3  # tonnes
 
-        elif self.mooring_type == "TLP":
+        elif self.mooring_type == "Tlp":
```
Comment (RHammond2) [inline ORBIT/phases/design/mooring_system_design.py:216]: I see how this is now annoying because there are many forms of standardization this could take.

**Review by RHammond2 — APPROVED** [review]
We could probably think a bit more on the capitalization of SemiTaut, TLP, etc., but I think it's good to go as-is.

File: ORBIT/phases/design/mooring_system_design.py (hunk 4 above)
Comment (nRiccobo) [reply ORBIT/phases/design/mooring_system_design.py:216]: Yeah... Never a one size fits all, heh.

**Pull request merged**